mods/cnc/maps/twist-of-fate/twist-of-fate.lua

```diff
@@ -0,0 +1,264 @@
+end
+
+WorldLoaded = function()
+
+	Camera.Position = DefaultCameraPosition.CenterPosition
+	Player1 = Player.GetPlayer("GDI")
+	EnemyAi = Player.GetPlayer("Nod")
+
+	if Difficulty == "hard" then
+		ProduceBuildingsDelay = 50
+		ProduceUnitsDelay = 10
+		ProductionCooldownSeconds = 30
+		MinHarvs = 3
+		AstkDelay = 180
+		NukeDelay = 900
+		ProcUpgrade = "AIHProcUpgrade"
+		EnemyAi.Resources = 3000
+		Player1.Cash = 3000
+		MCVReinf = { "mcv" }
+	end
+
+	if Difficulty == "normal" then
+		ProduceBuildingsDelay = 70
+		ProduceUnitsDelay = 15
+		ProductionCooldownSeconds = 55
+		MinHarvs = 3
+		AstkDelay = 220
+		NukeDelay = 1200
+		EnemyAi.Resources = 3000
+		Player1.Cash = 4000
+		ProcUpgrade = "AINProcUpgrade"
+		MCVReinf = { "mtnk", "mcv" }
+	end
+
+	if Difficulty == "easy" then
+		ProduceBuildingsDelay = 100
+		ProduceUnitsDelay = 35
+		ProductionCooldownSeconds = 85
+		MinHarvs = 2
+		AstkDelay = 250
+		NukeDelay = 1500
+		EnemyAi.Resources = 2000
+		Player1.Cash = 5000
+		MCVReinf = { "mtnk", "mtnk", "mcv", "mtnk" }
+		Actor137.Teleport(CPos.New(57,6))
+		Actor203.Destroy()
+	end
+
+	InitObjectives(Player1)
+	ClearPath = AddPrimaryObjective(Player1, "clear-path")
+	EliminateAllNod = AddPrimaryObjective(Player1, "eliminate-nod")
+
+	InitAI()
+	CheckObjectives()
+
+	Flare = Actor.Create("flare", true, { Owner = Player1, Location = DefaultFlareLocation.Location })
+
+	Trigger.AfterDelay(1,function()
+		AmbushTeam = Map.ActorsInBox(Map.CenterOfCell(CPos.New(46,5)), Map.CenterOfCell(CPos.New(60,53)), function(a) return a.Owner == EnemyAi and a.Type ~= "stnk" end)
+		Trigger.OnAllKilled(AmbushTeam, function()
+			Player1.MarkCompletedObjective(ClearPath)
+			Trigger.AfterDelay(50, function()
+				Trigger.AfterDelay(750, function()
+					Flare.Destroy()
+				end)
+				Media.PlaySpeechNotification(Player1, "Reinforce")
+				Reinforcements.Reinforce(Player1, MCVReinf, { CPos.New(56,2), waypoint0.Location }, 25, function(a)
+					a.Move(waypoint1.Location, 2)
+					a.Move(waypoint2.Location, 2)
+					a.Move(CPos.New(49,44), 2)
+					if a.HasProperty("Deploy") then
+						a.Move(CPos.New(48,51))
+						a.Deploy()
+					end
+				end)
+				Utils.Do(ClearPathCameras, function(c)
+					c.Destroy()
+				end)
+			end)
+		end)
+	end)
+
+	NodAirSupport = Actor.Create("Astk.proxy", true, { Owner = EnemyAi })
+	Trigger.AfterDelay(DateTime.Seconds(AstkDelay), function()
+		SendNodAirstrike(true)
+		Trigger.AfterDelay(DateTime.Seconds(20), function()
+			Media.DisplayMessage(UserInterface.Translate("air-strikes-intel-report"))
+			Trigger.AfterDelay(DateTime.Seconds(5), function()
+				CaptureCommCenter = AddSecondaryObjective(Player1, "capture-nod-communications-center")
+			end)
+		end)
+	end)
+
+	Trigger.OnKilledOrCaptured(NodOutPostCYard, function()
+		SendNodAirstrike(false)
+	end)
+
+	Trigger.OnAllKilled(NodSams, function()
+		AirSupport = Actor.Create("Astk.proxy", true, { Owner = Player1 })
+		Player1.MarkCompletedObjective(DestroySams)
```

Comment:
Though unlikely, a crash can occur here if a player doesn't first capture the airstrike HQ. Since the line references an objective ID that's only made after the HQ's capture...
```suggestion
		if DestroySams then
			Player1.MarkCompletedObjective(DestroySams)
		end
```